mint-auditor/src/db.rs

```diff
@@ -268,16 +285,27 @@ impl MintAuditorDb {
         Ok(decode(bytes)?)
     }
 
-    fn last_synced_block_index_impl(
-        &self,
-        db_txn: &impl Transaction,
-    ) -> Result<Option<u64>, Error> {
-        match db_txn.get(self.counts, &LAST_SYNCED_BLOCK_INDEX) {
-            Ok(bytes) => Ok(Some(key_bytes_to_u64(bytes))),
-            Err(lmdb::Error::NotFound) => Ok(None),
+    fn get_counters_impl(&self, db_txn: &impl Transaction) -> Result<Counters, Error> {
+        match db_txn.get(self.key_val, &COUNTERS_KEY) {
+            Ok(bytes) => Ok(decode(bytes)?),
+            Err(lmdb::Error::NotFound) => Ok(Counters::default()),
             Err(err) => Err(err.into()),
         }
     }
+
+    fn set_counters_impl<'env>(
+        &self,
+        counters: &Counters,
+        db_txn: &mut RwTransaction<'env>,
+    ) -> Result<(), Error> {
+        db_txn.put(
+            self.key_val,
+            &COUNTERS_KEY,
+            &encode(counters),
```

Comment:
Future thought: eventually we might want to look at a way to skip the encode steps by creating a way to feeding proto data directly into the db (skipping the rust encode/decode). Out of scope for this PR though.

Reply:
`encode` is Prost's method of taking a struct and converting it into protobuf bytes.